cmd/argocd/commands/app.go

```diff
@@ -1384,6 +1405,67 @@ func findandPrintDiff(ctx context.Context, app *argoappv1.Application, proj *arg
 	if diffOptions.local != "" {
 		localObjs := groupObjsByKey(getLocalObjects(ctx, app, proj, diffOptions.local, diffOptions.localRepoRoot, argoSettings.AppLabelKey, diffOptions.cluster.Info.ServerVersion, diffOptions.cluster.Info.APIVersions, argoSettings.KustomizeOptions, argoSettings.TrackingMethod), liveObjs, app.Spec.Destination.Namespace)
 		items = groupObjsForDiff(resources, localObjs, items, argoSettings, app.InstanceName(argoSettings.ControllerNamespace), app.Spec.Destination.Namespace)
+	} else if diffOptions.compareDesire && diffOptions.revision != "" {
```

Comment:
Option should also be valid for multi-source `diffOptions.revision != "" || len(diffOptions.revisions) > 0`

Reply:
I am not aware how to test this. Can you please help? I tried this but not getting any result while providing list of revsions.

Reply:
Also implemented this.